packages/fetchai/connections/p2p_libp2p/dht/dhtclient/dhtclient.go

```diff
@@ -105,6 +109,25 @@ func New(opts ...Option) (*DHTClient, error) {
 		return nil, errors.New("missing agent address")
 	}
 
+	// agent record is mandatory
+	if dhtClient.myAgentRecord == nil {
+		return nil, errors.New("missing agent record")
+	}
+
+	// check if the PoR is delivered for my public  key
```

Comment:
[STP2] peer will verify agent PoR before registering it the agent and saves it locally